### ISSUE TITLE
Introduce LiteralU32Expression

### ISSUE THREAD

**Issue by Yoric** [binast/ecmascript-binary-ast#32]
In practice, pretty much all literal numbers we encounter are within [|0, 2^32[|. Experience shows that introducing the following is a pretty good filesize win:

```idl
typedef (LiteralF64Expression or LiteralU32Expression)
  LiteralNumericExpression;

interface LiteralF64Expression {
  attribute double value;
}

interface LiteralU32Expression {
  attribute unsigned long value;
}
```

Any objection to this?

We could also make it `LiteralI32Expression` (with signed `long`s), with pretty much the same file size.

**Comment by ljharb** [comment]
Just to contrast with “never”: https://github.com/es-shims/es5-shim/blob/master/es5-shim.js#L1141, for example. Certainly it’s not common.

**Comment by Yoric** [comment]
Well, I can now replace the above *never* with *once*, but that doesn't change the trend :)

**Comment by ljharb** [comment]
I would suspect that the trend is actually much, much smaller numbers than 32 bit.

**Comment by Yoric** [comment]
Yes, the most common numbers I encounter are actually 0 and 1. So I believe that we should actually specialize even further for these numbers. But variable-length encoding gives us in a single byte (+generally 1 byte header) all numbers between -63 and +64, which should cover most cases, in two bytes (+ header) -8191 to +8192, which should cover most of the rest.

**Comment by Yoric** [comment]
For context, I have modified the binary format to transparently bias towards `i8`, then `i16`, then `i32`. Numbers whose value doesn't fit in a `i32` will take two more bytes, but experience shows that occurrences of this are nearly non-existent.

Another optimization being examined will automatically create a new pseudo-interface for common occurrences, including `0` and `1`. So, this issue is not necessary anymore for compression reasons.
